Dockerfile

```diff
@@ -1,24 +1,26 @@
 FROM ubuntu:16.04
 MAINTAINER Matthew Bruzek <Matthew.Bruzek@canonical.com>
 
-RUN useradd -m ubuntu -s /bin/bash
-RUN mkdir -p \
-    /home/ubuntu/.juju \
-    /home/ubuntu/builds \
-    /home/ubuntu/trusty \
-    /home/ubuntu/xenial
-
-RUN chown -R ubuntu:ubuntu /home/ubuntu
-
-VOLUME [ \
-    "/home/ubuntu/.juju", \
-    "/home/ubuntu/builds", \
-    "/home/ubuntu/trusty", \
-    "/home/ubuntu/xenial" \
-    ]
-
-COPY setup.sh /setup.sh
+ARG JUJU_USER=ubuntu
+
+ENV JUJU_HOME /home/$JUJU_USER/.juju
+ENV JUJU_REPOSITORY /home/$JUJU_USER/charms
+
+RUN useradd -m $JUJU_USER -s /bin/bash
+
+RUN mkdir -p $JUJU_HOME 
+RUN mkdir -p $JUJU_REPOSITORY
+
+RUN chown -R $JUJU_USER:$JUJU_USER $JUJU_HOME
+RUN chown -R $JUJU_USER:$JUJU_USER $JUJU_REPOSITORY
+
+VOLUME [ "$JUJU_HOME", "$JUJU_REPOSITORY" ]
+
+ADD setup.sh /setup.sh
```

Comment:
any reason you went with `ADD` over `COPY` here?  docker docs say `COPY` is preferred if you're not relying on `ADD` auto extraction capabilities:

https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/add-or-copy